### PR TITLE
[DPM-650] store affiliate id on auth

### PR DESCRIPTION
**Pull request by georgiypetrov**

### CODE REVIEW

File: auth/usecase.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 
 type UseCase interface {
 	ResolveUser(ctx context.Context, tmpToken string) (*models.User, error)
-	SignUp(ctx context.Context, user *models.User) (string, string, error) // returns: refreshToken, accessToken, error
+	SignUp(ctx context.Context, user *models.User, affiliateID string) (string, string, error) // returns: refreshToken, accessToken, error
```
Comment (georgiypetrov) [inline auth/usecase.go:10]: it may be worth adding affiliateID into models.User

File: auth/repository/postgres/user.go [hunk 2]
```diff
@@ -11,7 +12,9 @@ import (
 const (
 	selectUserCntByAccNameStmt = "SELECT count(*) FROM users WHERE account_name = $1"
 	selectUserByAccNameStmt    = "SELECT * FROM users WHERE account_name = $1"
+	selectAffIDByNameStmt      = "SELECT affiliate_id FROM affiliates WHERE account_name $1"
```
Comment (justefg) [inline auth/repository/postgres/user.go:15]: should be `account_name $1`

File: migrations/0008_affiliates_table.up.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE affiliates
+(
+    account_name VARCHAR(13) PRIMARY KEY UNIQUE,
```
Comment (justefg) [inline migrations/0008_affiliates_table.up.sql:3]: better `VARCHAR(13) REFERENCES users (account_name)`

File: migrations/0008_affiliates_table.up.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    affiliate_id VARCHAR NOT NULL
```
Comment (justefg) [inline migrations/0008_affiliates_table.up.sql:4]: maybe better to store `affiliate_id` in the `users` table?

File: auth/repository/postgres/user.go (hunk 2 above)
Comment (georgiypetrov) [reply auth/repository/postgres/user.go]: oops

File: migrations/0008_affiliates_table.up.sql (hunk 3 above)
Comment (georgiypetrov) [reply migrations/0008_affiliates_table.up.sql]: ok

File: migrations/0008_affiliates_table.up.sql (hunk 4 above)
Comment (georgiypetrov) [reply migrations/0008_affiliates_table.up.sql:4]: algys said it's better to create different table for affiliates.

**Review by justefg — APPROVED** [review]

**Pull request merged**